tests/garage/tf/core/test_cnn.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.core.cnn import cnn
+from tests.fixtures import TfGraphTestCase
+
+
+class TestCNN(TfGraphTestCase):
+    def setUp(self):
+        super(TestCNN, self).setUp()
+        self.obs_input = np.ones((2, 5, 4, 3))
+        input_shape = self.obs_input.shape[1:]  # height, width, channel
+        self.hidden_nonlinearity = tf.nn.relu
+
+        self._input_ph = tf.placeholder(
+            tf.float32, shape=(None, ) + input_shape, name="input")
+
+        self._output_shape = 2
+
+        # We build a default cnn
+        with tf.variable_scope("CNN"):
+            self.cnn = cnn(
+                input_var=self._input_ph,
+                output_dim=self._output_shape,
+                filter_dims=(3, 3, 3),
+                num_filters=(32, 64, 128),
+                stride=1,
+                name="cnn1",
+                hidden_nonlinearity=self.hidden_nonlinearity)
+
+        self.sess.run(tf.global_variables_initializer())
```

Comment:
Is this not done already in your fixture? Perhaps you called super too early?

Reply:
I didn't see this in `TfGraphTestCase`. Should I add this there?

Reply:
`TfGraphTestCase` is for testing graphs, where `tf.global_variables_initializer()` cannot be called in `setUp()`. I can create a new fixture for testing trivial things and initialize in `setUp()`.

Reply:
Is it possible to delete TfTestCase and only retain TfGraphTestCase? @jonashen 
Looks like creating a new graph and pass it to a session is relatively important when we have multiple tests in a test class, as in Anson's tests. And I can't find other tests that use TfTestCase.

We may as well call `tf.global_variables_initializer()` in setUp. Some algorithms test cases also use TfGraphTestCase. And algo.train() also calls `tf.global_variables_initializer()`. But it looks ok to call it twice in those cases.

Reply:
I defer to @CatherineSue 's judgement